README.rst

```diff
@@ -3,26 +3,47 @@ MeerKAT Science Data Processor Telescope State
 
 This is a client package that allows connection to the Redis database that
 stores telescope state information for the Science Data Processor of the
-MeerKAT radio telescope.
+MeerKAT radio telescope. This database is colloquially known as *telstate*.
 
 The Redis database acts as a key-value store. Each key is either a *sensor* or
 an *attribute*. A sensor has multiple timestamped values organised into an
 ordered set. An attribute (or *immutable* key) has a single value without a
 timestamp that is not allowed to change.
 
-The keys are strings and values are arbitrary Python objects stored as pickles.
-Keys can be accessed using attribute syntax or dict syntax.
+The keys are strings and the values are Python objects serialised via
+MessagePack_. Older versions of the database stored the values as pickles, and
+the package warns the user if that's the case. Keys can be retrieved from the
+telstate object using attribute syntax or dict syntax.
+
+.. _MessagePack: http://www.msgpack.org/
+
+Databases can be accessed via one of two backends: a Redis client backend
+that allows shared access to an actual Redis server over the network (or a
+simulated server via fakeredis) and a simplified in-memory backend for
+stand-alone access. Both backends support loading and saving a Redis snapshot
+in the form of an RDB dump file.
+
+It is possible to have multiple *views* on the same database (one per telstate
+instance). A view is defined as a list of *prefixes* acting as namespaces that
+group keys. When reading from the database, each prefix is prepended to the key
+in turn until a match is found. When writing to the database, the first prefix
+is prepended to the key. The first prefix therefore serves as the primary
+namespace while the rest are supplementary read-only namespaces.
 
 .. warning::
 
   **WARNING**: The standard warning about Python pickles applies. Never
-  retrieve data from an untrusted telescope state or connect to one over an
-  untrusted network.
+  retrieve data from an untrusted telstate database with values encoded as
+  pickles, or connect to such a database over an untrusted network.
+  This should only be an issue with older telstates stored in RDB files.
+
```

Comment:
Document the environment variable that controls whether pickles can be loaded. Disabling pickles should make it safe to connect to untrusted telstates.

Reply:
Done.